kong/clustering/rpc/manager.lua

```diff
@@ -372,24 +418,43 @@ function _M:connect(premature, node_id, host, path, cert, key)
   do
     local resp_headers = c:get_resp_headers()
     -- FIXME: resp_headers should not be case sensitive
-    if not resp_headers or not resp_headers["x_kong_rpc_capabilities"] then
-      ngx_log(ngx_ERR, "[rpc] peer did not provide capability list, node_id: ", node_id)
-      c:send_close() -- can't do much if this fails
-      goto err
-    end
 
-    local capabilities = resp_headers["x_kong_rpc_capabilities"]
-    capabilities = cjson_decode(capabilities)
-    if not capabilities then
-      ngx_log(ngx_ERR, "[rpc] unable to decode peer capability list, node_id: ", node_id,
-                       " list: ", capabilities)
+    if not resp_headers or not resp_headers["sec_websocket_protocol"] then
+      ngx_log(ngx_ERR, "[rpc] peer did not provide sec_websocket_protocol, node_id: ", node_id)
       c:send_close() -- can't do much if this fails
       goto err
     end
 
+    -- should like "kong.meta.v1"
+    local meta_rpc_call = resp_headers["sec_websocket_protocol"]
+
     local s = socket.new(self, c, node_id)
     s:start()
-    self:_add_socket(s, capabilities)
+    self:_add_socket(s)
+
+    ngx.timer.at(0, function(premature)
+      if premature then
+        return
+      end
+
+      local retry_count = 5
+
+      -- retry
+      for i = 1, retry_count do
+        local ok, err = self:_meta_call("control_plane", s, meta_rpc_call)
```

Comment:
We need to check if `meta_rpc_call` is `RPC_MATA_V1`. If not, then this CP can not handshake and connection must be terminated.

Reply:
done.